packages/dynamodb-cursor-based-pagination/README.md

```diff
@@ -0,0 +1,205 @@
+1. Items must be saved in such way that the range key ordination must represent the ordination of the pagination.
+
+Why do we need the second requirement? First we need to understand what cursor is. Cursor is like an edge identifier, with cursor we must be able to retrieve and locate that edge on your backend. In a DBB table with composite primary key, the sort key is a good choice to be our cursor.
+
+## Installation
+
+```
+npm install -S dynamodb-cursor-based-pagination
+```
+
+or
+
+```
+yarn add dynamodb-cursor-based-pagination
+```
+
+You also need install `aws-sdk` in your project because it is a peer dependency of this project.
+
+## How to Use
+
+```
+import { paginate } from 'dynamodb-cursor-based-pagination';
+```
+
+`paginate` is a method whose signature is:
+
+```ts
+type paginate<T = any> = ({
+  dynamoDBClient,
+  tableName,
+  hashKey,
+  hashKeyValue,
+  rangeKey,
+  index,
+  projectionExpression,
+  filterExpression,
+  filterAttributeNames,
+  filterAttributeValues,
+  beginsWith = '',
+  sort = 'DESC',
+  after,
+  first,
+  before,
+  last,
+}: {
+  dynamoDBClient: DynamoDBClient;
+  tableName: string;
+  hashKey: string;
+  hashKeyValue: string;
+  rangeKey: string;
+  beginsWith?: string;
+  index?: string;
+  projectionExpression?: string;
+  filterExpression?: string;
+  filterAttributeNames?: { [key: string]: string };
+  filterAttributeValues?: { [key: string]: any };
+  sort?: Sort;
+  after?: string;
+  before?: string;
+  first?: number;
+  last?: number;
+}) => Promise<{
+  edges: {
+    cursor: string;
+    node: T;
+  }[];
+  pageInfo: {
+    hasPreviousPage: boolean;
+    hasNextPage: boolean;
+    startCursor?: string | undefined;
+    endCursor?: string | undefined;
+  };
+  consumedCapacity: number | undefined;
+  count: number | undefined;
+  scannedCount: number | undefined;
+  lastEvaluatedKey: string | undefined;
+}>;
+```
+
+### Credentials
+
+You must have AWS credentials in your environment to use this package. The only permission needed is [dynamodb:Query](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/api-permissions-reference.html).
+
+If you don't have credentials in your environment, you may want provide them passing a [Credentials object](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html) to `credentials`:
+
+```ts
+import { Credentials } from 'aws-sdk';
+import { paginate } from 'dynamodb-cursor-based-pagination';
+
+const credentials = new Credentials({
+  accessKeyId: ...,
+  secretAccessKey: ...,
+  sessionToken: ...
+})
+
+paginate({
+  credentials,
+  ...
+})
+...
+```
+
+### DynamoDB Table Parameters
+
+The parameters `region`, `tableName`, `hashKeyName`, `hashKeyValue`, `rangeKeyName`, `indexName` are used to identify your DynamoDB table and the partition.
```

Comment:
`hashKeyName`, `rangeKeyName` and `indexName` don't exist anymore.

Reply:
Done.